modules/gapi/src/executor/gstreamingexecutor.cpp

```diff
@@ -853,16 +875,39 @@ class StreamingOutput final: public cv::gimpl::GIslandExecutable::IOutput
     }
 };
 
+struct IslandExecutableData {
+    std::shared_ptr<cv::gimpl::GIslandExecutable> island_exec;
+    std::string island_meta_info;
+};
+
+std::string buildIslandName(const ade::NodeHandle& op_nh,
+                            const cv::gimpl::GIslandModel::Graph& gim) {
+    std::string island_name = gim.metadata(op_nh).get<cv::gimpl::FusedIsland>().object->name();
+
+    std::string backend_name;
+    auto backend = gim.metadata(op_nh).get<cv::gimpl::FusedIsland>().object->backend();
+    std::string backend_impl_name = typeid(backend.priv()).name();
+
+    std::regex rgx("G([a-zA-Z0-9]+)BackendImpl");
+    std::smatch matches;
+    if (std::regex_search(backend_impl_name, matches, rgx)) {
+        GAPI_Assert(2ul == matches.size());
+        backend_name = matches[1].str();
+    }
+
+    return island_name + "_" + backend_name;
+}
+
 // This thread is a plain dumb processing actor. What it do is just:
 // - Reads input from the input queue(s), sleeps if there's nothing to read
 // - Once a full input vector is obtained, passes it to the underlying island
 //   executable for processing.
 // - Pushes processing results down to consumers - to the subsequent queues.
 //   Note: Every data object consumer has its own queue.
-void islandActorThread(std::vector<cv::gimpl::RcDesc> in_rcs,                // FIXME: this is...
-                       std::vector<cv::gimpl::RcDesc> out_rcs,               // FIXME: ...basically just...
-                       cv::GMetaArgs out_metas,                              // ...
-                       std::shared_ptr<cv::gimpl::GIslandExecutable> island, // FIXME: ...a copy of OpDesc{}.
+void islandActorThread(std::vector<cv::gimpl::RcDesc> in_rcs,  // FIXME: this is...
+                       std::vector<cv::gimpl::RcDesc> out_rcs, // FIXME: ...basically just...
+                       cv::GMetaArgs out_metas,                // ...
+                       IslandExecutableData island_data,       // FIXME: ...almost a copy of OpDesc{}.
```

Comment:
Not sure why you've had to alter the prototype here, I'd just put a string name as an extra parameter.

Reply:
I just can't cope in my mind with that that `islandActorThread` expects all arguments just to launch island code and island name parameter for something not transparent...
I think that from `islandActorThread` API it will be seen that `island_name` will be needed only for debug purposes, don't you think so?

Reply:
Thats correct but introducing a new type to deal with this tiny problem is overkill-ish

Reply:
Thanks, fixed!!